Sources/DatadogObjc/DDURLSessionDelegate+objc.swift

```diff
@@ -8,4 +8,27 @@ import Foundation
 import Datadog
 
 @objc
-public class DDNSURLSessionDelegate: DDURLSessionDelegate {}
+open class DDNSURLSessionDelegate: NSObject, URLSessionTaskDelegate, URLSessionDelegateProviding {
+    let swiftDelegate: DDURLSessionDelegate
+    public var ddURLSessionDelegate: DDURLSessionDelegate {
+        return swiftDelegate
+    }
+
+    @objc
+    override public init() {
+        swiftDelegate = DDURLSessionDelegate()
+    }
+
+    @objc
+    public init(additionalFirstPartyHosts: Set<String>) {
+        swiftDelegate = DDURLSessionDelegate(additionalFirstPartyHosts: additionalFirstPartyHosts)
+    }
+
+    open func urlSession(_ session: URLSession, task: URLSessionTask, didCompleteWithError error: Error?) {
+        swiftDelegate.urlSession(session, task: task, didCompleteWithError: error)
+    }
+
+    open func urlSession(_ session: URLSession, task: URLSessionTask, didFinishCollecting metrics: URLSessionTaskMetrics) {
+        swiftDelegate.urlSession(session, task: task, didFinishCollecting: metrics)
+    }
+}
```

Comment:
Even though it's a simple wrapper, we should add unit tests (like we do for other Objc <> Swift wrappers):
```swift
func testItForwardsCallsToSwiftDelegate() {
    let objcDelegate = DDNSURLSessionDelegate()
    let swiftDelegate = DDURLSessionDelegateMock()
    objcDelegate.swiftDelegate = swiftDelegate

    let sessionMock = URLSession.shared
    let taskMock = URLSessionTask.mockWith(request: .mockAny(), response: .mockAny())
    let metricsMock = URLSessionTaskMetrics.mockAny()
    let errorMock = ErrorMock()

    objcDelegate.urlSession(sessionMock, task: taskMock, didFinishCollecting: metricsMock)
    objcDelegate.urlSession(sessionMock, task: taskMock, didCompleteWithError: errorMock)

    XCTAssertTrue(swiftDelegate.calledDidFinishCollecting)
    XCTAssertTrue(swiftDelegate.calledDidCompleteWithError)
}
```
```swift
class DDURLSessionDelegateMock: DDURLSessionDelegate {
    var calledDidFinishCollecting = false
    var calledDidCompleteWithError = false

    override func urlSession(_ session: URLSession, task: URLSessionTask, didFinishCollecting metrics: URLSessionTaskMetrics) {
        calledDidFinishCollecting = true
    }

    override func urlSession(_ session: URLSession, task: URLSessionTask, didCompleteWithError error: Error?) {
        calledDidCompleteWithError = true
    }
}
```
Two tests for initialisation looks easy as well.